### PR TITLE
Randomize order of CDNs

### PR DESCRIPTION
**Pull request by mjackson**
Randomizes the order of the CDNs in the CDN list on a package page. See discussion in #723

### CODE REVIEW

**Comment by Haroenv** [comment]
Deploy preview ready!

Built with commit 95382358e607d67591a1f315494da5a1e1d071b0

https://deploy-preview-731--yarnpkg.netlify.com

**Comment by Haroenv** [comment]
They reorder on rerender now. Can you make the randomisation persist? 

Thanks, I think it makes sense ❤️

**Comment by mjackson** [comment]
> They reorder on rerender now. Can you make the randomisation persist?

Ah, sorry :P done!

File: js/src/lib/Details/Cdn.js [hunk 1]
```diff
@@ -47,16 +47,34 @@ const BundleRun = ({ name, version }) => (
...
+    };
+
+    this.order = Object.keys(this.cdns).sort(() => {
+      return Math.random() - 0.5;
+    });
+  }
+
   render() {
+    const items = this.order.map(key =>
+      React.createElement(this.cdns[key], {
```
Comment (xtuc) [inline js/src/lib/Details/Cdn.js]: You could also assign it to a variable with a capital name and use it as a component.

Comment (Daniel15) [reply js/src/lib/Details/Cdn.js]: ^ Yeah this is my preference, too

```
const items = this.order.map(key => {
 const CDNComponent = this.cdns[key];
  return (
    <CDNComponent
      key={key}
      ........
    />
  );
}
```

**Comment by Daniel15** [comment]
This looks good to me!

I had another idea that requires much more effort, but would be nicer long-term: Display them in order of usage, per-package. For example, if most people load `babel-standalone` via unpkg, display unpkg first. I think this would satisfy your use case, @mjackson 

This would require jsdelivr, bundle.run and unpkg to have an API that provides usage statistics per package. jsdelivr has an pretty powerful API for this (https://github.com/jsdelivr/data.jsdelivr.com#get-package-usage-stats) but  but I don't know if bundle.run or unpkg collect or provide similar stats.

File: js/src/lib/Details/Cdn.js (hunk 1 above)
Comment (mjackson) [reply js/src/lib/Details/Cdn.js]: JSX compiles down to `createElement`. Don't be afraid to use the JavaScript API when you've got JavaScript variables, guys 😅

**Comment by Haroenv** [comment]
Sorry for the confusion that this whole thing seems to be adding, I didn’t expect it. Thanks for participating @mjackson

File: js/src/lib/Details/Cdn.js (hunk 1 above)
Comment (Daniel15) [reply js/src/lib/Details/Cdn.js]: @mjackson - Consistency is good, though. Why use `React.createElement` at just one call site?

**Pull request merged**